opensearch-operator/pkg/builders/cluster.go

```diff
@@ -179,6 +179,21 @@ func NewSTSForNodePool(
 
 	image := helpers.ResolveImage(cr, &node)
 
+	var mainCommand []string
+	var com string
+	com = "yes | ./bin/opensearch-plugin install"
+	if len(cr.Spec.General.PluginsList) > 0 {
+		mainCommand = append(mainCommand, "/bin/bash", "-c")
+		for index, plugin := range cr.Spec.General.PluginsList {
+			fmt.Println(index, plugin)
+			com = com + " " + plugin
```

Comment:
this leads to a code injection risk, right? this allows passing in any string as a plugin name and it'll be passed to `/bin/bash -c` with no further checks. i.e. adding a plugin called `; echo "your code injection says hi!"; #` would probably work...?

Reply:
didnt understand the echo, can you explain it again ?

Reply:
I agree with @rursprung, there is a risk of injection. As such I suggest to put each plugin name in single quotes in the command string and also escape any single quote in the plugin name string.
Something along the lines of (not tested):
`com = com + " '" + strings.Replace(plugin, "'", "\\'", -1) + "'"`

Reply:
> didnt understand the echo, can you explain it again ?

if you just concatenate strings together and then execute it, it means that any attacker can control what you're executing. it's the epitome of a classical code injection bug.

so if you concatenate it together and somebody provides a plugin string called `; echo "your code injection says hi!"; #` (which, obviously, isn't a real plugin but an example of an injection) then you produce the following code:
```bash
yes | ./bin/opensearch-plugin install; echo "your code injection says hi!"; # ; ./opensearch-docker-entrypoint.sh
```

which means that:
* `yes | ./bin/opensearch-plugin install` will fail (due to no plugin being provided)
* `echo "your code injection says hi!"` will be executed (which in reality would be much more sinister than a simple echo...)
* `./opensearch-docker-entrypoint.sh` won't ever be called because it's been commented out (in a real attack this might not be the case: you could also imagine that the attack changes some security settings (e.g. adds a backdoor) and then lets opensearch start normally; to test, just replace `; #` with `; ./bin/opensearch-plugin install` (so that any additional addon still gets installed - if there's none it's just like the first step: it'll fail but it won't matter

on that note: i'd probably join the plugin call with the startup call with `&&`, not `;` as you probably don't want to start the cluster if you failed to install one or more plugins...?

Reply:
Make sense, will try to new PR today or tomorrow.
Thanks!